x-pack/plugins/security_solution/public/resolver/view/graph_controls.test.tsx

```diff
@@ -0,0 +1,207 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Simulator } from '../test_utilities/simulator';
+import { noAncestorsTwoChildren } from '../data_access_layer/mocks/no_ancestors_two_children';
+import { nudgeAnimationDuration } from '../store/camera/scaling_constants';
+import '../test_utilities/extend_jest';
+import { ReactWrapper } from 'enzyme';
+
+describe('graph controls', () => {
+  let simulator: Simulator;
+  let originEntityID: string;
+  let originNode: ReactWrapper;
+  const resolverComponentInstanceID = 'graph-controls-test';
+
+  beforeEach(async () => {
+    const {
+      metadata: { databaseDocumentID, entityIDs },
+      dataAccessLayer,
+    } = noAncestorsTwoChildren();
+
+    simulator = new Simulator({
+      dataAccessLayer,
+      databaseDocumentID,
+      resolverComponentInstanceID,
+    });
+    originEntityID = entityIDs.origin;
+  });
+
+  describe('when the graph controls load', () => {
+    it('should display all cardinal panning buttons and the center button', async () => {
+      await expect(
+        simulator.map(() => ({
+          westPanButton: simulator.westPanElement().length,
+          southPanButton: simulator.southPanElement().length,
+          eastPanButton: simulator.eastPanElement().length,
+          northPanButton: simulator.northPanElement().length,
+          centerButton: simulator.centerPanElement().length,
+        }))
+      ).toYieldEqualTo({
+        westPanButton: 1,
+        southPanButton: 1,
+        eastPanButton: 1,
+        northPanButton: 1,
+        centerButton: 1,
+      });
+    });
+
+    it('should display the zoom buttons and slider', async () => {
+      await expect(
+        simulator.map(() => ({
+          zoomInButton: simulator.zoomInElement().length,
+          zoomOutButton: simulator.zoomOutElement().length,
+          zoomSlider: simulator.zoomSliderElement().length,
+        }))
+      ).toYieldEqualTo({
+        zoomInButton: 1,
+        zoomOutButton: 1,
+        zoomSlider: 1,
+      });
+    });
+  });
+
+  describe('panning', () => {
+    const originalPositionStyle = { left: '746.93132px', top: '535.5792px' };
+    beforeEach(() => {
+      originNode = simulator.processNodeElements({ entityID: originEntityID });
+    });
+
+    describe('when the user has not interacted with panning yet', () => {
+      it("should show the origin node in it's original position", () => {
+        expect(originNode.getDOMNode()).toHaveStyle(originalPositionStyle);
+      });
+    });
+
+    describe('when the user clicks the west panning button', () => {
+      beforeEach(() => {
+        simulator.westPanElement().simulate('click');
+        simulator.runAnimationFramesTimeFromNow(nudgeAnimationDuration);
+      });
+
+      it('should show the origin node further left on the screen', async () => {
+        expect(originNode.getDOMNode()).toHaveStyle({ left: '796.93132px', top: '535.5792px' });
```

Comment:
I think this is fine to merge but long term I have some concerns:

This test relies on react to keep the same DOM node for the origin after the user clicks the pan element. If that component got a new DOM node for some reason then `originNode` would point to an old element that may not be in the DOM any more. 

Also this relies on the react code to update the style of the element in the same event loop cycle when the user clicks the panning button. 

I think we should use the pattern of calling `expect` on an async iterable. How about I make those changes after you merge?

Reply:
@michaelolo24 Here's a commit with my ideas: https://github.com/elastic/kibana/pull/74884/commits/3de7be9429a6f3fa801dbf6621b9232958e761ad

Reply:
Thanks, cherry-picked it in and updated the selectors based on our zoom meeting